internal/utils/persistence/src/main/java/org/eclipse/ditto/internal/utils/persistence/mongo/MongoClientWrapper.java

```diff
@@ -476,6 +484,18 @@ private static SSLContext createAndInitSslContext() throws NoSuchAlgorithmExcept
             return result;
         }
 
-    }
+        private void applyAwsIamRoleSettings() {
+            AwsCredentialsProvider credentialsProvider = getAwsCredentialsProvider();
+            AwsCredentials credentials = credentialsProvider.resolveCredentials();
+            MongoCredential credential = MongoCredential.createAwsCredential(
```

Comment:
Please declare all 3 variables as `final` - that's the "Ditto codestyle" to use `final` wherever possible.